test/test_filesystem_helper.cpp

```diff
@@ -477,16 +480,23 @@ TEST(TestFilesystemHelper, create_temp_directory)
   // bad names
   {
     if (is_win32) {
-      EXPECT_THROW(rcpputils::fs::create_temp_directory("illegalchar?"), std::system_error);
+      EXPECT_THROW(rcpputils::fs::create_temporary_directory("illegalchar?"), std::system_error);
     } else {
-      EXPECT_THROW(rcpputils::fs::create_temp_directory("base/name"), std::system_error);
+      std::filesystem::path tmpdir_with_slash_in_base;
+      EXPECT_NO_THROW(
+        tmpdir_with_slash_in_base = rcpputils::fs::create_temporary_directory("base/name"));
+      EXPECT_NO_THROW(std::filesystem::remove_all(tmpdir_with_slash_in_base));
```

Comment:
Note: In the new implementation, slashes in the base name is valid case and will be treated as path delimiters when creating folders.

Reply:
I'm not a huge fan of this behavior.  The base name is really the "template" for the name, and as such, it shouldn't be involved in creating additional subdirectories.  That seems like surprising behavior to me, and different than e.g. `mkdtemp` on Linux (which fails if the basename has a `/` in it).  So I'd suggest that if `base_name` contains a directory separator, we throw an exception.

Reply:
Ok, I added a throwing std::invalid_argument exception if base_name contains a directory separator.